AutoTag.Core/TV/ShowResults.cs

```diff
@@ -0,0 +1,116 @@
+
+    /// <summary>
+    /// Create simple ShowResult container with <see cref="SearchTv"/> base
+    /// </summary>
+    /// <param name="tvSearchResult"></param>
+    public ShowResults(SearchTv tvSearchResult)
+    {
+        TvSearchResult = tvSearchResult;
+    }
+
+    public static implicit operator ShowResults(SearchTv tv) => new(tv);
+
+    /// <summary>
+    /// Generates <see cref="ShowResults"/> list from any <see cref="SearchTv"/> enumerable 
+    /// </summary>
+    /// <param name="results">Result from tmdb client</param>
+    public static List<ShowResults> FromSearchResults(IEnumerable<SearchTv> results)
+    {
+        return results.Select(result => (ShowResults)result).ToList();
+    }
+
+    
+    /// <summary>
+    /// Add optional episode group to tv result
+    /// </summary>
+    /// <param name="episodeGroup">Episode group fetched from tmdb api</param>
+    public bool AddEpisodeGroup(TvGroupCollection episodeGroup)
+    {
+        if (!TryGenerateMappingTable(episodeGroup, out var mappingTable)) return false;
+
+        _episodeGroupMappingTable = mappingTable;
+        GroupCollection = episodeGroup;
+        return true;
+    }
+
+    /// <summary>
+    /// Try to retrieve episode mapping for episode group order
+    /// </summary>
+    /// <param name="seasonNumber">Season number as defined in episode group</param>
+    /// <param name="episodeNumber">Episode number as defined in episode group</param>
+    /// <param name="numbering">Matching episode number and season of "standard" order</param>
+    /// <returns>True if mapping exists, false if not</returns>
+    public bool TryGetMapping(int seasonNumber, int episodeNumber, out (int season, int episode)? numbering)
+    {
+        numbering = null;
+        if (_episodeGroupMappingTable?.TryGetValue((seasonNumber, episodeNumber), out var result) ?? false)
+        {
+            numbering = result;
+            return true;
+        }
+
+        return false;
+    }
+    
+    /// <summary>
+    /// Tries to generate mapping table between TMDB standard sorting of show
+    /// and the given Episode Group
+    /// </summary>
+    /// <param name="collection">Episode Group from TMDB</param>
+    /// <param name="parsedTable">Filled parsing table. Only filled when method returns true</param>
+    /// <returns>True if successful, false if not</returns>
+    private static bool TryGenerateMappingTable(TvGroupCollection collection,
+        out Dictionary<(int season, int episode), (int season, int episode)>? parsedTable)
+    {
+        parsedTable = new Dictionary<(int season, int episode), (int season, int episode)>();
+
+        foreach (var tvGroup in collection.Groups)
+        {
+            // determine season number
+            var sanitizedGroupName = tvGroup.Name.Trim().ToLower();
+            int? seasonNumber = null;
+
+            var seasonMatch = EpisodeRegex().Match(sanitizedGroupName);
```

Comment:
I'm not sure how reliable it will be to do the parsing this way, but is there really an alternative? This breaks in some cases (e.g. Season 04 Part 1 and 2 for [this](https://www.themoviedb.org/tv/160-teenage-mutant-ninja-turtles/episode_group/62f80aaa724de1007a0133c2) due to the duplicate season 4). It also breaks for any episode groups that aren't in English of course.

I don't think much can be done about this though, the subgroups don't have a set numerical season number so it all just depends on how the subgroups are named.

I'm not sure how common this is for episode groups on TMDB, I guess it would be fine to just add a disclaimer that it may not work in every case.